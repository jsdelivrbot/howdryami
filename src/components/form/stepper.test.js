/* eslint-disable */

import React from 'react';
import Stepper from './stepper';


describe('Test that the stepper', () => {
  let stepperProps;

  beforeEach(() => {
    stepperProps = {
      stepList: [{ key:'FOO', value: 'Foo' }, { key:'BAR', value: 'Bar' }],
      value:"FOO",
      fieldName:"acme",
      onUpdate: () => {},
    }
  });

  test('renders without crashing', () => {
    const output = shallow(
      <Stepper  { ...stepperProps } />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  })

  test('calls function on step up', () => {

    const stepper = shallow(
      <Stepper { ...stepperProps } />
    );

    expect(stepper.state().currentStepIndex).toEqual(0);
    stepper.instance().stepOnce(1);
    expect(stepper.state().currentStepIndex).toEqual(1);
  })
});

