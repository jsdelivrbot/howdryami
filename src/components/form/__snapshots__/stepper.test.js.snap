// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the stepper renders without crashing 1`] = `
<View
  className="stepper"
>
  <Text
    className="stepper__label"
  />
  <View
    className="stepper__wrapper"
  >
    <Button
      className="stepper__button stepper__button--left"
      onClick={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button/REGULAR"
    />
    <Text
      className="stepper__value"
    >
      Foo
       
    </Text>
    <Button
      className="stepper__button stepper__button--right"
      onClick={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button/REGULAR"
    />
  </View>
</View>
`;
